imx943_evk/mimx94398/a55: all applications fail to build: fatal error: fsl_elec_spec.h: No such file or directory

**Describe the bug**

All applications targeting `imx943_evk/mimx94398/a55` fail to build on current main.

Reported by @hakehuang while running the NXP full regression for #113811 (the failure is unrelated to that PR's pinctrl change): https://github.com/zephyrproject-rtos/zephyr/pull/113811#issuecomment-5030628333

```
In file included from modules/hal/nxp/mcux/mcux-sdk-ng/devices/i.MX/i.MX943/periph/PERI_ADC.h:177,
                 from modules/hal/nxp/mcux/mcux-sdk-ng/devices/i.MX/i.MX943/MIMX94398/MIMX94398_ca55.h:89,
                 from modules/hal/nxp/mcux/mcux-sdk-ng/devices/i.MX/i.MX943/MIMX94398/fsl_device_registers.h:17,
                 from zephyr/soc/nxp/imx/imx9/imx943/a55/./soc.h:10,
                 from zephyr/boards/nxp/imx943_evk/board.c:9:
modules/hal/nxp/mcux/mcux-sdk-ng/devices/i.MX/i.MX943/MIMX94398/MIMX94398_ca55_COMMON.h:5194:10: fatal error: fsl_elec_spec.h: No such file or directory
 5194 | #include "fsl_elec_spec.h"
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
```

**Root cause analysis**

- The MCUX SDK 26.06.00 sync zephyrproject-rtos/hal_nxp#760 added `#include "fsl_elec_spec.h"` to `MIMX94398_ca55_COMMON.h`.
- The header does exist in hal_nxp at `mcux/mcux-sdk-ng/devices/i.MX/i.MX943/MIMX94398/drivers/fsl_elec_spec.h`, but that directory is not on the include path when building Zephyr for imx943 a55, so every build of this target breaks.

@JiafeiPan confirmed the breakage on the hal_nxp side and offered to submit a fix: https://github.com/zephyrproject-rtos/hal_nxp/pull/760#issuecomment-5043666609

@ZhaoxiangJin pointed out a candidate fix already in flight in #113641 (`modules/hal_nxp/mcux/mcux-sdk-ng/device/device.cmake`): https://github.com/zephyrproject-rtos/zephyr/pull/113641/changes/359744f7c1ad8cf7155d9ce628093367c5824dad — please coordinate so only one fix lands.

**To Reproduce**

```
west build -b imx943_evk/mimx94398/a55 samples/hello_world
```

with hal_nxp at or after commit 2ba4f14239 ("mcux: sdk-ng: update device/i.mx to 26.06.00").

**Expected behavior**

imx943_evk/mimx94398/a55 applications build successfully.

**Impact**

All builds for imx943_evk/mimx94398/a55 are broken (build blocker for this target, including CI regression runs).

**Environment**

- OS: any
- Toolchain: Zephyr SDK
- Zephyr: main, hal_nxp revision containing zephyrproject-rtos/hal_nxp#760 (SDK 26.06.00 sync)

--- a/modules/hal_nxp/mcux/mcux-sdk-ng/device/device.cmake
+++ b/modules/hal_nxp/mcux/mcux-sdk-ng/device/device.cmake
@@ -141,16 +141,6 @@ endif()
 # Load device files
 mcux_add_cmakelists(${mcux_device_folder})
 
-# i.MX943 Cortex-A ships ca55/exception.c, a weak putc/puts helper that pulls in
-# the SDK debug console (fsl_debug_console.h). That header is not part of
-# hal_nxp and Zephyr provides its own console, so drop the file from the build.
-if(CONFIG_SOC_MIMX94398 AND CONFIG_CPU_CORTEX_A)
-  mcux_project_remove_source(
-    BASE_PATH ${SdkRootDirPath}
-    SOURCES devices/i.MX/i.MX943/MIMX94398/ca55/exception.c
-  )
-endif()
-
 # Workaround for fsl_flexspi_nor_boot link error, remove the one in SDK, use the Zephyr file.
 if(CONFIG_MCUX_COMPONENT_device.boot_header)
 

--- a/west.yml
+++ b/west.yml
@@ -213,7 +213,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 6c77523dfdfc5a1ecefbafb31feee6499a5de8ec
+      revision: pull/776/head
       path: modules/hal/nxp
       groups:
         - hal

--- a/west.yml
+++ b/west.yml
@@ -213,7 +213,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: ba28ac2057aa97c5a18780cecf93e54ccf927fae
+      revision: 1fad84dd567809ba69cc519c7bc58bc6458d1654
       path: modules/hal/nxp
       groups:
         - hal

